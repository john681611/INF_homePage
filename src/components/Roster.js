import React from 'react'

const Roster = (props) => {
  return (
    <section id="roster" class="section">
      <div class="section__inner">
        <h2>Roster</h2>
        {props.squads.map(squad =>
          <span>
            <h3>{squad.name}</h3>
            <table class="table table-inverse">
              <thead>
                <tr>
                  <th>Name</th>
                  <th class="no-mobile">Rank</th>
                  <th class="no-mobile">Position</th>
                  <th>Role</th>
                  <th class="no-mobile">Enlisted</th>
                </tr>
              </thead>
              <tbody>
                {props.members.filter(member => member.squad === squad.id).map(member =>
                  <tr>
                    <td>
                      {member.name}
                    </td>
                    <th class="no-mobile" scope="row">
                      {member.rank}
                    </th>
                    <td class="no-mobile">
                      {member.pos}
                    </td>
                    <td>
                      {member.role}
                    </td>
                    <td class="no-mobile">
                      {member.joined}
                    </td>
                  </tr>
                )}
              </tbody>
            </table>
            <hr />
          </span>
        )}
      </div>
    </section>
  )
}

export default Roster