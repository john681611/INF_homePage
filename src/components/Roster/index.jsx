import React from 'react'

const Roster = (props) => {
  return (
    <section id="roster" className="section">
      <div className="section__inner">
        <h2>Roster</h2>
        {props.squads.map(squad =>
          <span  key={squad.name}>
            <h3>{squad.name}</h3>
            <table className="table table-inverse">
              <thead>
                <tr>
                  <th>Name</th>
                  <th className="no-mobile">Rank</th>
                  <th className="no-mobile">Position</th>
                  <th>Role</th>
                  <th className="no-mobile">Enlisted</th>
                </tr>
              </thead>
              <tbody>
                {props.members.filter(member => member.position.includes(squad.name)).map(member =>
                  <tr key={member.nickname}>
                    <td className="member-name">
                      {member.name || member.nickname}
                    </td>
                    <th className="no-mobile member-remark" scope="row">
                      {member.remark}
                    </th>
                    <td className="no-mobile member-position">
                      {member.position}
                    </td>
                    <td className="member-role">
                      {member.role}
                    </td>
                    <td className="no-mobile member-joined">
                      {member.joined}
                    </td>
                  </tr>
                )}
              </tbody>
            </table>
            <hr />
          </span>
        )}
      </div>
    </section>
  )
}

export default Roster