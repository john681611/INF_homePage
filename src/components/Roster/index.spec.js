import React from 'react';
import { shallow } from 'enzyme';
import Roster from './';

describe('Roster', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Roster
      members={[]}
      squads={[]}
      />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render 1 section', () => {
    expect(wrapper.find('section').length).toEqual(1);
  });

  it('should render Squad Sub sections', () => {
    wrapper.setProps({ squads: [{
        name: 'Alpha'
    },
    {
        name: 'Bravo'
    }
    ], members: [] });
    const squad = wrapper.find('h3');
    expect(squad.length).toEqual(2);
    expect(squad.at(0).text()).toEqual('Alpha');
    expect(squad.at(1).text()).toEqual('Bravo');
  });

  it('should render one Squad and 2 members', () => {
    wrapper.setProps({ squads: [{
        name: 'Alpha'
    }
    ], members: [
        {
            position: 'Alpha 1',
            name: 'Bob',
            nickname: 'BobKey',
            remark: 'My hovercraft Is full of eels',
            role: 'Janitor',
            joined: 2018
        },
        {
            position: 'Alpha 2',
            nickname: 'Sid',
            remark: 'My hovercraft Is full of frogs',
            role: 'Basket Weaver',
            joined: 2019
        }
    ]});

    const names = wrapper.find('.member-name');
    expect(names.length).toEqual(2);
    expect(names.at(0).text()).toEqual('Bob');
    expect(names.at(1).text()).toEqual('Sid');

    const remarks = wrapper.find('.member-remark');
    expect(remarks.length).toEqual(2);
    expect(remarks.at(0).text()).toEqual('My hovercraft Is full of eels');
    expect(remarks.at(1).text()).toEqual('My hovercraft Is full of frogs');

    const position = wrapper.find('.member-position');
    expect(position.length).toEqual(2);
    expect(position.at(0).text()).toEqual('Alpha 1');
    expect(position.at(1).text()).toEqual('Alpha 2');

    const role = wrapper.find('.member-role');
    expect(role.length).toEqual(2);
    expect(role.at(0).text()).toEqual('Janitor');
    expect(role.at(1).text()).toEqual('Basket Weaver');

    const joined = wrapper.find('.member-joined');
    expect(joined.length).toEqual(2);
    expect(joined.at(0).text()).toEqual("2018");
    expect(joined.at(1).text()).toEqual("2019");
  });
});