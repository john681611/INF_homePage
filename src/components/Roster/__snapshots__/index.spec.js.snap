// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster should render correctly 1`] = `ShallowWrapper {}`;
